### PR TITLE
extensions/kde-neon-6: fix XDG_CONFIG_DIRS

### PR DESCRIPTION
**Pull request by soumyaDghosh**
- [x] Have you followed the [guidelines for contributing](https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: extensions/desktop/kde-neon-6/launcher-specific [hunk 1]
```diff
@@ -15,6 +15,12 @@ append_dir LD_LIBRARY_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/blas"
 # lapack
 append_dir LD_LIBRARY_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/lapack"
 
+# KConfig uses QStandardPaths:GenericConfigLocation to get the config dir paths
+# and checks for the kdeglobals file there for theming. The kdeglobals file is
+# stored in $SNAP_REAL_HOME/.config folder.
+
+prepend_dir XDG_CONFIG_DIRS "$SNAP_REAL_HOME/.config"
```
[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply extensions/desktop/kde-neon-6/launcher-specific:22]: https://github.com/canonical/snapd/pull/14172

Already fixed that problem

[the remark this answers is not in the document]
Comment (sergiusens) [reply extensions/desktop/kde-neon-6/launcher-specific:22]: we need to add an assumes for the snap being built in the PackageService depending on the extension being used (or do it through the extension itself), unless this does not break in any bad way if access is denied

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply extensions/desktop/kde-neon-6/launcher-specific:22]: No it just leaves kirigami based apps use the default fallback and not the one defined by the user. No breakages.

[the remark this answers is not in the document]
Comment (lengau) [reply extensions/desktop/kde-neon-6/launcher-specific:22]: Thanks! Sounds good then.

**Review by lengau — APPROVED** [review]

**Comment by ScarlettGatelyMoore** [comment]
Thank you, except this PR needs to be done here: https://invent.kde.org/neon/snap-packaging/snapcraft-desktop-integration

**Comment by soumyaDghosh** [comment]
> Thank you, except this PR needs to be done here: https://invent.kde.org/neon/snap-packaging/snapcraft-desktop-integration

Let's merge it here until it's used from here, else we'll not be able to use the change.

**Comment by soumyaDghosh** [comment]
@sergiusens @lengau I think we can merge this. The Code Coverage issue is not related to this I think.

**Comment by soumyaDghosh** [comment]
> > @sergiusens @lengau I think we can merge this. The Code Coverage issue is not related to this I think.
> 
> @soumyaDghosh can you please also do a pr here for this https://invent.kde.org/neon/snap-packaging/snapcraft-desktop-integration

Very sorry, I forgot about this. It's done!

https://invent.kde.org/neon/snap-packaging/snapcraft-desktop-integration/-/merge_requests/2

**Comment by soumyaDghosh** [comment]
@sergiusens @lengau kindly merge this...

**Comment by castle-giant** [comment]
I'm running into this and it's very frustrating.  Is there anything I can do to help getting this merged? @soumyaDghosh

**Comment by soumyaDghosh** [comment]
> I'm running into this and it's very frustrating.  Is there anything I can do to help getting this merged? @soumyaDghosh 

This MR should be merged. If you want a quick work around for existing snaps, then create a symlink of .config/kdeglobals in 
`snap/snap_name/current/.config/kdeglobals`

**Comment by castle-giant** [comment]
Adding the symlink didn't help, thanks anyway though!  I'll have to be patient.

**Comment by soumyaDghosh** [comment]
> Adding the symlink didn't help, thanks anyway though!  I'll have to be patient.

Which snap is it? Also what's the exact path did you put the file?

**Comment by castle-giant** [comment]
This is for kasts.  I ran '~/snap/kasts/current/.config$ ln --symbolic ~/.config/kdeglobals'

**Comment by soumyaDghosh** [comment]
> This is for kasts.  I ran '~/snap/kasts/current/.config$ ln --symbolic ~/.config/kdeglobals'

So is there, in `snap/kasts/current/.config`, a file named kdeglobals showing up?

**Comment by castle-giant** [comment]
Yes!On Dec 6, 2024, at 22:14, সৌম্যদীপ ঘোষ ***@***.***> wrote:
So is there, in snap/kasts/current/.config, a file named kdeglobals showing up?

**Comment by soumyaDghosh** [comment]
> Yes!On Dec 6, 2024, at 22:14, সৌম্যদীপ ঘোষ ***@***.***> wrote: So is there, in snap/kasts/current/.config, a file named kdeglobals showing up? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

Working for me, Let's not hijack this thread and ping me in matrix.

image

**Pull request merged**